gitlint/git.py

```diff
@@ -129,18 +132,36 @@ def from_local_repository(repository_path):
                 '_cwd': repository_path
             }
 
+            sha_list = sh.git("rev-list", refspec, **sh_special_args).split()
+
             # Get info from the local git repository
             # https://git-scm.com/docs/pretty-formats
-            commit_msg = sh.git.log("-1", "--pretty=%B", **sh_special_args)
-            commit_author_name = sh.git.log("-1", "--pretty=%aN", **sh_special_args)
-            commit_author_email = sh.git.log("-1", "--pretty=%aE", **sh_special_args)
-            # %aI -> ISO 8601-like format, while %aI is strict ISO 8601, it seems to be less widely supporte
-            commit_date_str = sh.git.log("-1", "--pretty=%ai", **sh_special_args)
-            commit_parents = sh.git.log("-1", "--pretty=%P", **sh_special_args).split(" ")
-            commit_is_merge_commit = len(commit_parents) > 1
-
-            # changed files in last commit
-            changed_files_str = sh.git("diff-tree", "--no-commit-id", "--name-only", "-r", "HEAD", **sh_special_args)
+            for sha in sha_list:
```

Comment:
I like this a lot. Don't know what I hadn't thought about using a single call to git with a single ```--pretty``` line in the past.